assets/js/components/notifications/SetupErrorNotification.test.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * SetupErrorNotification component tests.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	render,
+	createTestRegistry,
+	provideSiteInfo,
+} from '../../../../tests/js/test-utils';
+import {
+	VIEW_CONTEXT_MAIN_DASHBOARD,
+	VIEW_CONTEXT_SPLASH,
+} from '../../googlesitekit/constants';
+import { withNotificationComponentProps } from '../../googlesitekit/notifications/util/component-props';
+import SetupErrorNotification from './SetupErrorNotification';
+
+const SETUP_ERROR_NOTIFICATION = 'setup_error';
+
+describe( 'SetupErrorNotification', () => {
+	let registry;
+
+	const NotificationWithComponentProps = withNotificationComponentProps(
+		SETUP_ERROR_NOTIFICATION
+	)( SetupErrorNotification );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		provideSiteInfo( registry, {
+			setupErrorRedoURL: '#',
+			setupErrorCode: 'access_denied',
+			setupErrorMessage:
+				'Setup was interrupted because you did not grant the necessary permissions',
+		} );
+	} );
+
+	it( 'should display the notification when there is a permission error during setup', async () => {
+		const { getByText, waitForRegistry } = render(
+			<NotificationWithComponentProps />,
+			{
+				registry,
+				viewContext: VIEW_CONTEXT_SPLASH,
+			}
+		);
+
+		await waitForRegistry();
+
+		expect(
+			getByText( /oops! there was a problem during set up/i )
+		).toBeInTheDocument();
+	} );
+
+	// Skipped in the meantime until notifications API testing infrastructure is set in place,
+	// currently this can't be tested as banner itself does not handle the checks and queue when it will be shown.
+	// @TODO enbale test once notifications API is included in tests.
+	// eslint-disable-next-line jest/no-disabled-tests
+	it.skip( 'should not display the notification when permission error happens outside the setup screen', async () => {
+		const { getByText, waitForRegistry } = render(
+			<NotificationWithComponentProps />,
+			{
+				registry,
+				viewContext: VIEW_CONTEXT_MAIN_DASHBOARD,
+			}
+		);
+
+		await waitForRegistry();
+
+		expect(
+			getByText( /oops! there was a problem during set up/i )
+		).not.toBeInTheDocument();
+	} );
```

Comment:
My question here is should we leave a skipped test in the suite or remove it and add a IB item to re-add it in the ticket where it is possible to do so?

Reply:
It is following the approach done in few initial notifications that were refactored, where tests are also skipped. There is an open issue for infra improvement after which all skipped tests will be revisited

Reply:
@zutigrm We now have a simple framework to test notifications after #9273 was merged. So we shouldn't skip tests anymore.

Reply:
@jimmymadon Nice, thanks, will update tests then

Reply:
@zutigrm Speaking of tests, @aaemnnosttv has introduced a nice pattern to solely test the `checkRequirements` function which is generally what is unique and isn't tested as part of the generic notifications API tests. If applicable, then we should follow this pattern when refactoring the `active` / `not active` states based solely on the `checkRequirements` callback. The changes are in this [PR](https://github.com/google/site-kit-wp/pull/9684) and not yet merged.

Reply:
Looks elegant, although in this specific scenario, it didn't apply correctly, since `checkRequirements` does not accept context as a parameter, and notification here is rendered based on the context, so I will need to use the previous infra added to render notifications in tests